Pause button

JeLuF opened the pull request:
![image](https://user-images.githubusercontent.com/5852422/206882341-07b01b2f-b3f1-4175-8a27-45ffffb65389.png)

Pause button that will prevent new render tasks from being started. Running tasks will not be interrupted, they can complete.

https://discord.com/channels/1014774730907209781/1014774732018683928/1050977439179874375
https://discord.com/channels/1014774730907209781/1014774732018683928/1051282461419642991

cmdr2 commented:
Thanks!

I agree with Patrice about this going next to the "stop all" button. Should also rename to "Pause All", for consistency.

cmdr2 commented:
Nicely done btw

madrang commented on ui/media/js/engine.js:
```diff
@@ -437,6 +437,9 @@
          */
         async post(url, timeout=-1) {
+	    if (typeof pauseClient != 'undefined' && pauseClient===true && typeof resumeClient === 'function') {
+	        await resumeClient()
```
The event callback onIdle is now async in main.js
https://github.com/cmdr2/stable-diffusion-ui/pull/643

Calling `await resumeClient()` from onIdle will now do just the same and would keep all the UI code inside main.js

JeLuF commented:
Removed changes from engine.js, moved the pause check into onIdle() as @madrang suggested.
Moved the pause button next to the Stop All button, renamed to Pause All.
![image](https://user-images.githubusercontent.com/5852422/206908110-c1691659-6c82-4f7e-8612-f2a2bf5c60b2.png)

madrang commented:
Look great.
I made a small change to reduce that small delay when resuming.
https://github.com/cmdr2/stable-diffusion-ui/pull/649

cmdr2 commented:
Thanks!
